tensorboard/defs/defs.bzl

```diff
@@ -149,3 +149,29 @@ def tf_ng_module(assets = [], **kwargs):
         angular_assets = assets,
         **kwargs
     )
+
+def tf_res_digest_suffixer(name, resources, template, out):
+    """Query parameter suffixer for a resource in a template.
+
+    In order to facilitate resource caching, the macro suffixes resource declaration in
+    a template file by replacing it with one with query parameter,
+    "?_file_md5=[len_8_truncated_md5]".
+
+    For example, if a template is "index.html" that contains content like below,
+
+        <script src="index.js"></script>
+
+    For resources of ["index.js"], it will be replaced with:
+
+        <script src="index.js?_file_md5=486c31fc"></script>
+    """
+
+    native.genrule(
+        name = name,
+        srcs = [template] + resources,
+        outs = [out],
+        cmd = "$(execpath //tensorboard/defs:res_digest_suffixer) $(SRCS) > $@",
+        tools = [
```

Comment:
Should be `exec_tools`, as this tool is meant to be run on the [platform
on which Bazel actions run, not the one on which Bazel itself runs.][1]
Cf. #4031.

[1]: https://docs.bazel.build/versions/master/platforms.html